File: backend/app/core/system_properties.py
SYSTEM_PROPERTY_PREFIX = "MAP_"
CAD_PROPERTY_NAMESPACE = "CAD_"


def system_property(name: str) -> str:
    return f"{SYSTEM_PROPERTY_PREFIX}{name}"


def cad_property(name: str) -> str:
    return system_property(f"{CAD_PROPERTY_NAMESPACE}{name}")


APP_SOURCE_LAYER_PROPERTY = system_property("sourceLayer")
APP_SOURCE_LAYER_KIND_PROPERTY = system_property("sourceLayerKind")
CAD_LAYER_PROPERTY = cad_property("layer")
CAD_TYPE_PROPERTY = cad_property("type")

LEGACY_SOURCE_LAYER_PROPERTY = "sourceLayer"
LEGACY_CAD_LAYER_PROPERTY = "CAD_layer"

RESTORE_LAYER_FIELDS = (
    APP_SOURCE_LAYER_PROPERTY,
    CAD_LAYER_PROPERTY,
    LEGACY_SOURCE_LAYER_PROPERTY,
    LEGACY_CAD_LAYER_PROPERTY,
)

File: frontend/lib/gis/systemProperties.ts
export const SYSTEM_PROPERTY_PREFIX = "MAP_";
export const CAD_PROPERTY_NAMESPACE = "CAD_";

export function systemProperty(name: string) {
  return `${SYSTEM_PROPERTY_PREFIX}${name}`;
}

export function cadProperty(name: string) {
  return systemProperty(`${CAD_PROPERTY_NAMESPACE}${name}`);
}

export const mapProperties = {
  sourceLayer: systemProperty("sourceLayer"),
  sourceLayerId: systemProperty("sourceLayerId"),
  sourceLayerKind: systemProperty("sourceLayerKind"),
  sourceDatasetId: systemProperty("sourceDatasetId"),
  sourceDatasetLayerId: systemProperty("sourceDatasetLayerId"),
  sourceCategory: systemProperty("sourceCategory"),
  georeferenceStatus: systemProperty("georeferenceStatus"),
  cadLayer: cadProperty("layer"),
  cadType: cadProperty("type"),
  cadText: cadProperty("text"),
  cadTextHeight: cadProperty("text_height"),
  cadRotation: cadProperty("rotation")
} as const;
